src/librbd/api/PoolMetadata.cc

```diff
@@ -61,6 +65,27 @@ int PoolMetadata<I>::set(librados::IoCtx& io_ctx, const std::string &key,
     return r;
   }
 
+  if (update_pool_timestamp) {
+    auto now = ceph_clock_now();
+    std::string cmd =
+      R"({)"
+        R"("prefix": "config set", )"
+        R"("who": "global", )"
+        R"("name": "rbd_config_pool_override_update_timestamp", )"
```

Comment:
It looks very clever though just a little hackish to me.

Alternatively I think we could add "config refresh <who> [key_prefix]"  mon command, and the librbd would send "config refresh rbd_config" here. It will not work for older cluster until backported though.

Anyway, I am fine with your approach if you don't like the "config refresh".

Reply:
It needs to work from the API as well (dashboard). I think adding a second step makes it more error prone to forget that your changes weren't actually applied.

Reply:
... I think I misunderstood your original proposal. I thought you were talking about adding something to the `rbd` CLI. I'm not sure how that would work if there was no global override for the key. Right now it just sends all applicable config values to the client when updated and the client determines if it was actually updated before invoking the observer callback. To implement something like this, we would need a new message (or flag on the MConfig message) to instruct `config::set_mon_vals` that key XYZ was updated but we don't know the value.

Reply:
Why a second step? I meant instead of introducing a "strange" (to users) variable and sending to the mon "set this strange rbd variable to a new value" we would send a newly introduced command "refresh all config variables starting with 'rbd_' prefix" (there was a bug in my previous message: `s/rbd_config/rbd_/`). And this would trigger the config observer handler to refresh the image. The problem as I said though that this command would be available in new releases only, which is probably a stopper if you want to backport this.